### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by anbraten** [woodpecker-ci/woodpecker#2170] [Announcements]
This is a general roadmap for the project covering the most requested epics. They are sorted in some kind of priority (considered by maintainers), but this is not forced.

# General roadmap
- [x] #1451
- [ ] Add more pipeline events https://github.com/woodpecker-ci/woodpecker/discussions/3425
  - [x] Cron #8 #934 
  - [x] Support manually triggering a pipeline #83 #240 
  - [x] PR close #286
  - [x] Release (type: draft, pre, stable; action: created, updated, deleted) #3226  
  - [ ] comments (create, delete, update) / #814 
  - [ ] label (added, removed, issue, PR)
  - [ ] issues (created, updated, closed, reopened)
- [ ] well known #2159  
- [x] Autoscaler #999
- [x] BringYourOwnAgent #267
- [x] non root images #1077
- [ ] pipeline compile step #2254 
- [ ] enhance plugins so they can **easily** be used for caching #2295 
- [ ] multiple forges #138
- [x] UI configured forges #425
- [x] Admin panel #345
- [ ] UI plugins which can be used by plugins #2287 
- [ ] Add support for HA setup #2304  
- [x] Decide to use Swagger docs or Graphql #292  
- [ ] #84
- [x] repo extensions (external secrets, registries, config compilers) #783 
- [x] deprecation warnings #922

**Comment by anbraten** [comment]
As 1.0 was released now this is my plan for issues I want to work on next:
- [x] #1572, #276, #922
- [x] #286 
- [ ] think about caching #2295  
- [ ] #138
- [x] maybe #267 
- [ ] some kind of advanced pipeline config maybe by sth like #2254 #2286

**Comment by anbraten** [comment]
# Breaking changes
- [x] `pipeline` to `steps` #2180 
- [x] `branches` to `when.branch` #2180 
- [x] `platform` to `labels` #2480 
- [x] step `needs` keyword / DAG #1860
- [ ] pre-processor syntax #2473 
- [ ] remove custom clone step property: `clone` #2474
- [ ] drop `detach: true` in favor of "normal" service https://woodpecker-ci.org/docs/next/usage/services#detachment
- [x] handle new events (filters) https://github.com/woodpecker-ci/woodpecker/discussions/2174

**Reply by anbraten** [reply]
As we are kinda stuck atm with https://github.com/woodpecker-ci/woodpecker/pull/2476 I am thinking again of reverting the breaking `pipeline` => `steps` (support both versions for now) to release the next `1.1.0`. In the next step we could then consider adding a project setting to select one of the woodpecker config compilers (v1,v2, ... or even a custom one #2254) with the compilers being some kind of sidecar in the future.

**Reply by runephilosof-karnovgroup** [reply]
I am considering deploying `next` in production because I need https://github.com/woodpecker-ci/woodpecker/pull/2163.
It would be great with a 1.x release, so I can avoid that.

**Reply by runephilosof-karnovgroup** [reply]
Or a 2.x release with the current breaking changes.

**Comment by 6543** [comment]
I'm against removing detatched:

There is no equivalent ... services wont do as they start before while detatched can start and use e.g. artifacts build by a step before

**Reply by qwerty287** [reply]
I'd rather implement something like https://github.com/woodpecker-ci/woodpecker/issues/2495 so the service is only active as long it's needed.

**Reply by qwerty287** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/3411 will allow this
